distributed/protocol/serialize.py

```diff
@@ -233,36 +239,16 @@ def serialize(x, serializers=None, on_error="message", context=None):
     if isinstance(x, Serialized):
         return x.header, x.frames
 
-    if type(x) in (list, set, tuple, dict):
-        iterate_collection = False
-        if type(x) is list and "msgpack" in serializers:
-            # Note: "msgpack" will always convert lists to tuples
-            #       (see GitHub #3716), so we should iterate
-            #       through the list if "msgpack" comes before "pickle"
-            #       in the list of serializers.
-            iterate_collection = ("pickle" not in serializers) or (
-                serializers.index("pickle") > serializers.index("msgpack")
-            )
-        if not iterate_collection:
-            # Check for "dask"-serializable data in dict/list/set
-            iterate_collection = check_dask_serializable(x)
-
     # Determine whether keys are safe to be serialized with msgpack
-    if type(x) is dict and iterate_collection:
+    if type(x) is dict:
         try:
             msgpack.dumps(list(x.keys()))
         except Exception:
             dict_safe = False
         else:
             dict_safe = True
 
-    if (
-        type(x) in (list, set, tuple)
-        and iterate_collection
-        or type(x) is dict
-        and iterate_collection
-        and dict_safe
-    ):
+    if type(x) in (list, set, tuple) or type(x) is dict and dict_safe:
```

Comment:
Note that this change makes #4641 unnecessary, because this PR is effectively setting `iterate_collection=True` in all cases.  This is great for HLG serialization on the scheduler, because we need the serialization code path to detect existing `Serialized` objects within the elements of a serialized task.

With that said, I do want to make sure that we want this to be the **default** behavior for all objects.  That is, do we want to be diving in to every element when the tuple length is 1000+?